src/http/request.rs

```diff
@@ -75,6 +75,13 @@ impl<'a> Request<'a> {
         token: Option<&str>,
         proxy: Option<&str>,
     ) -> Result<ReqwestRequestBuilder> {
+        // Build the multipart first, as to keep future size small.
+        let multipart = if let Some(multipart) = self.multipart {
+            Some(multipart.build_form().await?)
+        } else {
+            None
+        };
+
```

Comment:
I'm confused here. The function call doesn't capture any locals, so why would it matter when it gets awaited? Or does it capture all of `self`?

Reply:
The function call doesn't use any of the locals, but the state machine for `Request::build` still needs to store the locals. By moving this back, we don't have to store the `path`/`builder`/`headers` in the state machine as there are no awaits after this.